validator/client/propose.go

```diff
@@ -150,6 +162,146 @@ func (v *validator) ProposeBlock(ctx context.Context, slot types.Slot, pubKey [4
 	}
 }
 
+// This is a routine to propose altair compatible beacon blocks.
+func (v *validator) proposeBlockAltair(ctx context.Context, slot types.Slot, pubKey [48]byte) {
+	if slot == 0 {
+		log.Debug("Assigned to genesis slot, skipping proposal")
+		return
+	}
+	ctx, span := trace.StartSpan(ctx, "validator.proposeBlockAltair")
+	defer span.End()
+
+	lock := mputil.NewMultilock(fmt.Sprint(iface.RoleProposer), string(pubKey[:]))
+	lock.Lock()
```

Comment:
It looks to me like a slashing protection thing. It prevents multiple callers using the same `pubKey` to propose block at the same time (until the proposed block is written in slashing protection db) to prevent against double proposal

Reply:
yep it will prevent another proposer with the same public key from calling this function at the same time